Build Eror: 'stddef.h' file not found

There appears to be some implicit assumptions being made about the build environment.

Although it's somewhat obvious, the docs don't mention a hard dependency on `weechat-dev` or similar packages. However, with that installed, `stddef.h` (required by `weechat-plugin.h`) is not being found.

I'm not exactly sure what it's referring to:

```
xxx@professorx:~/Development/community/weechat-matrix-rs$ fdfind stddef.h /|grep -vE 'flatpak|media'
/usr/include/linux/stddef.h
/usr/src/linux-headers-4.19.0-11-common/include/linux/stddef.h
/usr/src/linux-headers-4.19.0-12-common/include/linux/stddef.h
/usr/lib/gcc/x86_64-linux-gnu/6/include/stddef.h
/usr/lib/gcc/x86_64-linux-gnu/8/include/stddef.h
/usr/lib/gcc/avr/5.4.0/include/stddef.h
/usr/lib/llvm-7/include/c++/v1/stddef.h
/usr/src/linux-headers-4.19.0-11-common/include/uapi/linux/stddef.h
/usr/src/linux-headers-4.19.0-12-common/include/uapi/linux/stddef.h

```

It might be worth making mention of non-rust build dependencies in the README.

Build with both:

```
rust-weechat/weechat-sys on  master [⇡] is 📦 v0.4.0 via 🦀 v1.48.0                                                                                  ❯ grep -A3 builder.header build.rs
    builder = builder.header(file)
        .clang_arg("-I/usr/include/")                                                                                                                
        .clang_arg("-I/usr/lib/gcc/x86_64-linux-gnu/8/include/");                                                                                    
                                                                    
                                                                                                                                                     
rust-weechat/weechat-sys on  master [⇡] is 📦 v0.4.0 via 🦀 v1.48.0                                                                                 
❯ cargo build                                                                                                                                        
   Compiling weechat-sys v0.4.0 (/home/david/Development/community/rust-weechat/weechat-sys)
    Finished dev [unoptimized + debuginfo] target(s) in 4.04s    
```

Build without /usr/include:
```
rust-weechat/weechat-sys on  master [⇡] is 📦 v0.4.0 via 🦀 v1.48.0 took 4s
❯ sed -i -e 's/.clang_arg("-I\/usr\/include\/")//' build.rs
         
rust-weechat/weechat-sys on  master [!⇡] is 📦 v0.4.0 via 🦀 v1.48.0
❯ cargo build
   Compiling weechat-sys v0.4.0 (/home/david/Development/community/rust-weechat/weechat-sys)
    Finished dev [unoptimized + debuginfo] target(s) in 4.01s
```

Build without either:
```
rust-weechat/weechat-sys on  master [!⇡] is 📦 v0.4.0 via 🦀 v1.48.0
❯ sed -i -e 's/.clang_arg("-I\/usr\/lib\/gcc\/x86_64-linux-gnu\/8\/include\/")//' build.rs                                                           

rust-weechat/weechat-sys on  master [!⇡] is 📦 v0.4.0 via 🦀 v1.48.0
❯ cargo build                                                                                                                                        
   Compiling weechat-sys v0.4.0 (/home/david/Development/community/rust-weechat/weechat-sys)                                                         
error: failed to run custom build command for `weechat-sys v0.4.0 (/home/david/Development/community/rust-weechat/weechat-sys)`                      

Caused by:
  process didn't exit successfully: `/home/david/Development/community/rust-weechat/target/debug/build/weechat-sys-40fddab25177a659/build-script-build` (exit code: 101)
  --- stderr
  src/wrapper.h:1:10: fatal error: 'weechat/weechat-plugin.h' file not found                                                                         
  src/wrapper.h:1:10: fatal error: 'weechat/weechat-plugin.h' file not found, err: true                                                              
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found, err: true                                                                      
  thread 'main' panicked at 'Unable to generate bindings: ()', weechat-sys/build.rs:72:61                                                            
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                      

rust-weechat/weechat-sys on  master [!⇡] is 📦 v0.4.0 via 🦀 v1.48.0 took 2s                                                                        
❯ grep -A3 builder.header build.rs                                                                                                                   
    builder = builder.header(file)

        ;
```
So it does seem like the first include path is unnecessary. The second does seem to be required, for whatever reason.

GCC 8 is my default compiler:
```
❯ cpp -v /dev/null -o /dev/null
Using built-in specs.
COLLECT_GCC=cpp
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 8.3.0-6' --with-bugurl=file:///usr/share/doc/gcc-8/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-8 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 8.3.0 (Debian 8.3.0-6) 
COLLECT_GCC_OPTIONS='-E' '-v' '-o' '/dev/null' '-mtune=generic' '-march=x86-64'
 /usr/lib/gcc/x86_64-linux-gnu/8/cc1 -E -quiet -v -imultiarch x86_64-linux-gnu /dev/null -o /dev/null -mtune=generic -march=x86-64
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/8/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/8/include
 /usr/local/include
 /usr/lib/gcc/x86_64-linux-gnu/8/include-fixed
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/8/:/usr/lib/gcc/x86_64-linux-gnu/8/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/8/:/usr/lib/gcc/x86_64-linux-gnu/
LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/8/:/usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/8/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-linux-gnu/8/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-E' '-v' '-o' '/dev/null' '-mtune=generic' '-march=x86-64'

```

The `ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/8/../../../../x86_64-linux-gnu/include"` looks like a likely culprit, perhaps.

I checked several other Debian 10 machines, and each of them include that message in `cpp -v` output. Given that, it seems possible that this is an issue on Debian generally. That said, I've not previously encountered this issue when compiling other -sys crates.

I had the same problem.

```
error: failed to run custom build command for `weechat-sys v0.4.0 (https://github.com/poljar/rust-weechat#fafc7e7e)`

Caused by:
  process didn't exit successfully: `/home/arouene/Projects/weechat-matrix-rs/target/release/build/weechat-sys-8f3a560455ba992d/build-script-build` (exit status: 101)
  --- stderr
  /usr/include/weechat/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found
  /usr/include/weechat/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found, err: true
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', /home/arouene/.cargo/git/checkouts/rust-weechat-4239b35d876f5f87/fafc7e7/weechat-sys/build.rs:70:61
```
I am on Fedora 34, I needed to install `clang`, `clang-libs` was already installed but not `clang` :man_shrugging: , `gcc-c++` was also already installed without success.

On CentOS 7.9.2009, installed `clang`, `clang-libs`, `llvm`, `llvm-libs`, seems `gcc-c++` is also installed: `gcc-c++-4.8.5`.

Getting the same error:

```
error: failed to run custom build command for `weechat-sys v0.4.0 (https://github.com/poljar/rust-weechat#fafc7e7e)`

Caused by:
  process didn't exit successfully: `/home/myuser/git/weechat-matrix-rs/target/debug/build/weechat-sys-1a657337347b7df7/build-script-build` (exit status: 101)
  --- stderr
  /usr/include/weechat/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found
  /usr/include/weechat/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found, err: true
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found
  src/weechat-plugin.h:29:10: fatal error: 'stddef.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', 
```

--- a/README.md
+++ b/README.md
@@ -43,10 +43,20 @@ path to that header:
 
     export WEECHAT_PLUGIN_FILE=/usr/include/weechat/weechat-plugin.h
 
+Use the same WeeChat API headers as the WeeChat binary that will load the
+plugin. If WeeChat reports an API mismatch while loading `matrix.so`, rebuild
+after installing the matching development package or setting
+`WEECHAT_PLUGIN_FILE`. Set `WEECHAT_BUNDLED=1` only when you deliberately want
+to build against the bundled fallback header, for example in a controlled test
+or package build.
+
 If `bindgen` reports that `stddef.h`, `libclang`, or `libLLVM` is missing, check
 that the `clang` command-line tools, C/C++ compiler headers, and matching
 `libclang` package are installed. On minimal distributions the library package
 alone may not provide the compiler include paths that `bindgen` needs.
+Use a coherent `clang`/`libclang`/`libLLVM` set from the same toolchain; copying
+only one shared library from another package or SDK can leave `bindgen` loading
+an incompatible LLVM library.
 
 After the dependencies are installed the plugin can be compiled with:
 
@@ -185,6 +195,24 @@ The inactivity delay defaults to five minutes and can be changed with:
 
        /set matrix-rust.look.smart_filter_delay 300000
 
+# Room Status And Verification
+
+The `buffer_modes` bar item shows Matrix room state. Encrypted rooms show
+`matrix-rust.look.encrypted_room_sign`; encrypted rooms that still contain
+unverified devices also show `matrix-rust.look.encryption_warning_sign`. Public
+rooms show `matrix-rust.look.public_room_sign`, and busy rooms show
+`matrix-rust.look.busy_sign`.
+
+Interactive verification requests can be accepted, confirmed, or cancelled from
+the room or verification buffer:
+
+       /verification accept
+       /verification confirm
+       /verification cancel
+
+The plugin renders in-room SAS verification requests and shows the emoji or
+decimal short authentication string when it is ready.
+
 # Matrix Spaces
 
 Room buffers expose Matrix parent Spaces as WeeChat local variables. Switch to a
@@ -206,6 +234,12 @@ example, to prefix Matrix room names with their first parent Space:
 Remove that prefix again with:
 
        /unset buflist.format.name
+
+Matrix rooms can belong to more than one Space. WeeChat buffers cannot be shown
+as the same room under multiple server-like parents at the same time, so the
+plugin exposes all known parent Spaces as metadata instead of moving room
+buffers into a fake hierarchy. Matrix Space buffers use a `+` short-name prefix;
+ordinary rooms keep the usual room/channel naming rules.
 
 # Storage
 

